LogicalTypes.md

```diff
@@ -366,3 +366,27 @@ optional group my_map (MAP_KEY_VALUE) {
 }
 ```
 
+### Unions
+
+A Union type annotates data stored as a Group.
+It describes the different possible types under the same field name.
+The names of the fields in the annotated Group are not important in such a case, but
+as a convention the type names are used.
+All fields of the Group must be optional and exactly one is defined for each instance of the group.
+If more than one is defined the behavior is undefined and may changed depending on the projection applied.
+A Union can not contain null but can be null itself if in an optional field.
+
+// Union<String, Integer, Boolean> (nullable union of either String, Integer or Boolean)
+optional group my_union (Union) {
+  optional binary string (UTF8);
+  optional int32 integer;
+  optional boolean bool;
+}
+
+// Union<String, Integer, Boolean> (required union of either String, Integer or Boolean)
```

Comment:
I'd change the wording here, too. The union isn't required, one of the branches is required to be non-null.